internal/deploy/bootstrap.go

```diff
@@ -63,16 +65,97 @@ func Bootstrap(
 		manifests = append(manifests, []byte(wildCardRoleAndAssignment)...)
 	}
 
-	if err := applyManifests(
+	manifestObjs, err := parseManifests(k8s, manifests, dryRun)
+	if err != nil {
+		return false, err
+	}
+	if err = PatchDeploymentWithInKcpModeFlag(manifestObjs, isInKcpMode); err != nil {
+		return false, err
+	}
+	err = applyManifests(
 		ctx, k8s, manifests, applyOpts{
-			dryRun, force, defaultRetries, defaultInitialBackoff},
-	); err != nil {
+			dryRun, force, defaultRetries, defaultInitialBackoff}, manifestObjs,
+	)
+	if err != nil {
 		return false, err
 	}
 
 	return hasKyma(string(manifests))
 }
 
+func PatchDeploymentWithInKcpModeFlag(manifestObjs []ctrlClient.Object, isInKcpMode bool) error {
+	var deployment *appsv1.Deployment
+	for _, manifest := range manifestObjs {
+		manifestJSON, err := getManifestJSONForDeployment(manifest)
+		if err != nil || manifestJSON == nil {
+			return err
```

Comment:
If `manifestJSON == nil` is evaluated to true there won't be an error and this function will return nil.